### PR TITLE
feature: collapse search queries script

### PR DESCRIPTION
**Pull request by aaryanpunia**

### CODE REVIEW

**Review by densumesh — APPROVED** [review]

File: docker/collapse-query-script/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,15 @@
+# Use an official Python runtime as a parent image
```
Comment (densumesh) [inline docker/collapse-query-script/Dockerfile:1]: Create a CI action for building the docker image

**Review by densumesh — CHANGES_REQUESTED** [review]

File: .github/workflows/push-collapse-query-script.yml [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+
+      - name: Docker meta
+        id: meta
+        uses: docker/metadata-action@v5
+        with:
+          # list of Docker images to use as base name for tags
+          images: |
+            trieve/clickhouse-collapse-query-script
+          tags: |
+            type=raw,latest
```
Comment (cdxker) [inline .github/workflows/push-collapse-query-script.yml:43]:
```suggestion
            type=raw,latest
            type=sha
```
Add this so we can tag the releases to the git commit

**Review by cdxker — CHANGES_REQUESTED** [review]

**Review by cdxker — APPROVED** [review]

**Review by skeptrunedev — CHANGES_REQUESTED** [review]
Add a README to `docker/collapse-query-script` which describes how this works 

I can't understand it just reading the code at glance

**Review by skeptrunedev — CHANGES_REQUESTED** [review]
README needs update to reflect time bound on collapse now

**Review by skeptrunedev — COMMENTED** [review]
Need to actually test this locally and make sure it works as expected then I think it's good to merge

**Review by densumesh — APPROVED** [review]

**Pull request merged**